lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/runner/Runner.scala

```diff
@@ -232,13 +232,19 @@ class Runner(
       case None =>
         runtimeVersionManager.withEngineAndRuntime(engineVersion) {
           (engine, runtime) =>
-            val ensoLauncher = Option(System.getenv(Runner.LAUNCHER_ENV_NAME))
-            val requiresJVMRunner =
-              ensoLauncher.exists(_.equals("shell")) || runSettings.jvmMode
-            if (requiresJVMRunner) {
+            if (runSettings.jvm.isDefined) {
+              val javaHome = runSettings.jvm.get
+              val javaExec = if (javaHome.isDefined) {
+                new JavaExecCommand(
+                  javaHome.get.resolve("bin").resolve("java").toString,
+                  javaHome.map(_.toString)
+                )
+              } else {
+                JavaExecCommand.forRuntime(runtime)
+              }
```

Comment:
```suggestion
              val javaExec = javaHome.map(home =>
                new JavaExecCommand(
                  home.resolve("bin").resolve("java").toString,
                  javaHome.map(_.toString)
                )
              ).getOrElse(JavaExecCommand.forRuntime(runtime))
```
I wouldn't say this is fancy.